### PR TITLE
Android 13 media permissions

### PR DESCRIPTION
**Pull request by irfano**
This updates media permissions for Android 13 compatibility. 

[**Granular media permissions**](https://developer.android.com/about/versions/13/behavior-changes-13#granular-media-permissions)
`READ_EXTERNAL_STORAGE` is used for accessing photos, videos, and audio files. After upgrading `targetSdk` to 33, we need to use specific permissions, `READ_MEDIA_IMAGES`, `READ_MEDIA_VIDEO`, and `READ_MEDIA_AUDIO`.

- I've used granular permissions for `>=API 33`. [READ_EXTERNAL_STORAGE](https://developer.android.com/reference/android/Manifest.permission#READ_EXTERNAL_STORAGE) is still used for older APIs. `READ_MEDIA_IMAGES` and `READ_MEDIA_VIDEO` are different permissions on the code, but they appear as single permission, "Photos and videos". So, I requested both `READ_MEDIA_IMAGES` and `READ_MEDIA_VIDEO` even though just photos are needed.
- I've also updated [WRITE_EXTERNAL_STORAGE] (https://developer.android.com/reference/android/Manifest.permission#WRITE_EXTERNAL_STORAGE) usage, which has no effect for devices higher than API 29.
- pdf files are not listed on the media picker on devices that use API 30 or newer. It's not introduced with API 33. Opened an [issue](https://github.com/wordpress-mobile/WordPress-Android/issues/18231).
- Some strings were wrong. They're also updated in this PR. (`Storage and Storage` → `Photos and videos`)

|before denied|after denied - all media|after denied - images|after denied - audio|
|-|-|-|-|
|||||

To test:
Please perform all tests using a device with API 33 or higher and a device with an API version lower than 33.

**Media Picker - Image - Choose from device**
1. Launch the JP app.
2. Create a blog post.
3. Add an Image block.
4. Tap on the block and tap "Choose from device".
5. Allow permission and add an image.
6. Ensure the image is added without any issues.
7. Disable the permission from the system settings. 
8. Repeat 1-6, but deny the permission this time. Ensure the message direct you to the system settings to allow the permission. 
9. Enable the permission on system settings 
10. Add an image and ensure it's added as expected.

**Media Picker - Image - Other options**
1. Repeat "**Media Picker - Image - Choose from device**", but this time, try selecting other options instead of "Choose from device".

**Media Picker - Video**
Repeat **Media Picker - Image** tests with the video.

**Media Picker - Audio**
Repeat **Media Picker - Image** tests with the audio.

**Media Browser**
1. Launch the JP app.
2. Ensure you have all types of files on your media browser. (image, video, audio, and a file such as pdf)
3. Navigate to "My Site → Media".
4. Ensure all files on your device are listed.
5. Tap an image.
6. On the media detail screen, tap the save button at top of the screen.
7. Ensure the file is downloaded to your device.
8. Repeat 1-7 for other types of files.

**Story**
> **Warning**
> Story feature can be tested once https://github.com/Automattic/stories-android/pull/731 is ready. 
1. Launch the JP app.
2. Tap the green floating button at the bottom of the screen.
3. Tap "Story post". If you see a popup, tap Create Post button to continue.
4. Allow permissions and ensure you can see your media on the device.
5. Select a media and post the story.
6. Create another post and take a photo this time.
7. Ensure you can post the story by taking a photo or video.
8. Put the app in the background.
9. Go to the system settings and disable the permissions.
10. Go back to the create story screen. Ensure denied permission message is shown.
11. To ensure it works as expected for all cases, try to allow and disable some permissions and test different variations (like allow audio, deny camera, deny images). 

## Regression Notes
1. Potential unintended areas of impact
All screens that need storage permission.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested all cases manually.

3. What automated tests I added (or what prevented me from doing so)
Updated existing tests.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by ravishanker** [comment]
👋 @irfano - Nice work!  All the steps are working, except **Take a Photo** or **Video** is asking for **Storage** permission, and that leads to nowhere in system settings where it can be enabled!  Looked through _System Settings -> Apps_ and through _Privacy -> Permissions Manager_.   Happening on both JP and WP.

**Comment by fluiddot** [comment]
> 👋 @irfano - Nice work! All the steps are working, except **Take a Photo** or **Video** is asking for **Storage** permission, and that leads to nowhere in system settings where it can be enabled! Looked through _System Settings -> Apps_ and through _Privacy -> Permissions Manager_. Happening on both JP and WP.

Seems the **Storage** permission is asked due to the logic when taking a photo from the editor:
https://github.com/wordpress-mobile/WordPress-Android/blob/8a07a96023f52d6d8cc6b6af0b8d4bd1fd9b6ec9/libs/editor/src/main/java/org/wordpress/android/editor/gutenberg/GutenbergEditorFragment.java#L284-L287
https://github.com/wordpress-mobile/WordPress-Android/blob/8a07a96023f52d6d8cc6b6af0b8d4bd1fd9b6ec9/libs/editor/src/main/java/org/wordpress/android/editor/gutenberg/GutenbergEditorFragment.java#L774-L775

It relies on [`PermissionUtils.checkAndRequestCameraAndStoragePermissions` ](https://github.com/wordpress-mobile/WordPress-Utils-Android/blob/a6483fc2f5eecfd653280dd96e5a2025fe4bc5d8/WordPressUtils/src/main/java/org/wordpress/android/util/PermissionUtils.java#L79-L91), which as far as I checked, it tries to access `android.permission.WRITE_EXTERNAL_STORAGE` permission. 

@irfano @ravishanker Maybe a potential workaround would be to apply a similar approach like [this one](https://github.com/wordpress-mobile/WordPress-Android/pull/18183/files#diff-5aae92a41306a6b5a8aa97f3f41405b5c2ec6513b4bf39f8bcd6f65be17ee3ceR681-R687) on the `PermissionUtils` class, WDYT?
https://github.com/wordpress-mobile/WordPress-Android/blob/bb26d70d882088d1db51027ee65a18d2173f2640/WordPress/src/main/java/org/wordpress/android/ui/media/MediaBrowserActivity.java#L1006-L1012

**Comment by irfano** [comment]
Thank you so much for the help @fluiddot! I thought the change was required on gutenberg-mobile but it was on Utils library as you said.
I opened a [PR on Utils library](https://github.com/wordpress-mobile/WordPress-Utils-Android/pull/126), and it fixed the camera issue on this PR.

**Comment by irfano** [comment]
@ravishanker, I updated the Utils library, and the camera permission issue should be fixed now.

**Comment by ravishanker** [comment]
> @ravishanker, I updated the Utils library, and the camera permission issue should be fixed now.

Yes, working now.

Published a new version of Utils 3.5.0, please update it.

**Review by ravishanker — APPROVED** [review]

**Review by pachlava — APPROVED** [review]
@irfano 👋 

Just leaving a note that `mobile-ui-testing-squad` was pinged for a review because the PR removes the write permission from two UI tests-related file. This does not seem to affect the execution negatively, so from our side, it's good to go as well 🙂

**Comment by AjeshRPai** [comment]
Hey @irfano 

Nice work. 🙌🏼 . I have tested the app on Android 12, Pixel, and Android 13, Samsung. 

While the positive scenario(all permissions granted) works in all the cases on both the devices, I found a behavior in Android 13, Samsung. I am not sure if it's right. More details below

| Permissions  |  Media → + icon → Choose from device Screen  | 
|---|---|
| <br>Audio and Music Permission - Allowed  ✅  </br> <br> Photos and Videos Permission - Dont Allow ❌  </br> |  Screenshot 2023-03-29 at 9 42 13 PM |  
| <br> Audio and Music Permission -  Don't allow ❌  </br> <br>Photos and Videos Permission - Allow ✅ </br>  |  WhatsApp Image 2023-03-29 at 9 47 45 PM  | 

I haven't worked on the permissions for some time, so I am not sure what the issue is or whether this is to be expected.

**Comment by irfano** [comment]
> While the positive scenario(all permissions granted) works in all the cases on both the devices, I found a behavior in Android 13, Samsung. I am not sure if it's right. More details below

🤔 Thank you for testing, @AjeshRPai! Can you see photos even though permission for the "Photos & videos" is not allowed? I tested it on both the emulator and the Android 13 Samsung device but couldn't reproduce your case. Can you write me testing instructions to reproduce it?

**Comment by AjeshRPai** [comment]
Hey Irfan 👋🏼 

I can see some photos even when the permission `Photos and Videos` is not allowed and `Audio and Music` is allowed. When I am not allowing the permission `Audio and Music`, I cannot see anything even when `Photos and Video` permission is allowed

Steps to reproduce 
- Deny the permissions `Photos and Videos` and `Audio and Music`
- Go to Dashboard → Media  → + icon → Choose from device  
- Verify that the permission dialog is open → Deny the permission 
- Put the app in background 
- Go to the System settings →  Allow `Photos and Videos` and don't allow `Audio and Music`
- Go to the app →  Notice that no photos or images are shown on Screen 
- Go to the System settings  → Change `Photos and Videos`to Don't allow  and allow `Audio and Music`
- Go to the app → Notice that some photos are shown

Here the `Audio and Music` seems to be the breaker in showing the Photos, My instinct is that in Android 13 either, 
- We need the Audios and Music permission to access photos and video. 
   - In that case, we need to update the messaging to include `Audio and Music` on screen from the current implementation in this PR to the one similar in this [PR](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/pull/63) raised by @0nko .
- We don't need the Audios and Music permission, but we are incorrectly checking the status of Audio and Music permission before showing the photos and videos on media picker.

| Current permission message  | Permission message shown in this [PR](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android/pull/63)   | 
|---|---|
|   WhatsApp Image 2023-03-29 at 9 47 45 PM| 228355568-2ba9f8e5-cebc-4e1a-9f28-3497bf616b3b |   

@irfano I will spend some additional time checking whether I can reproduce the scenario I have described [here](https://github.com/wordpress-mobile/WordPress-Android/pull/18183#issuecomment-1488930831) . I will make sure to let you know with a comment.

File: WordPress/src/main/java/org/wordpress/android/ui/photopicker/MediaPickerLauncher.kt [hunk 1]
```diff
@@ -192,7 +187,8 @@ class MediaPickerLauncher @Inject constructor(
             primaryDataSource = DEVICE,
             availableDataSources = setOf(),
             canMultiselect = canMultiselect,
-            requiresStoragePermissions = true,
+            requiresPhotosVideosPermissions = false,
```
Comment (AjeshRPai) [inline WordPress/src/main/java/org/wordpress/android/ui/photopicker/MediaPickerLauncher.kt]: ❓I believe this is the function that's being called when you navigate from Media  → + Icon  → Choose from Device.  Is the `requiresPhotosVideosPermissions` having correct value here?.  

I did a quick run and test after changing this `requiresPhotosVideosPermissions` to `true`. I was able to see the photos in the Scenario 

- Allowed Photos and Video & Dont allow Audio and Music - I can see the photos and videos
- Dont allow Photos and Video & Allow Audio and Music - I can't see any photos and video, Permission message is shown
- Dont allow both permissions - Permission message is shown

**Comment by AjeshRPai** [comment]
@irfano 

I have tested the app again, I can reproduce this issue. ✅  I have ensured that I am on the correct branch, and the Android version is 13. 

📹  Please find below the video capture of this issue.

https://user-images.githubusercontent.com/17463767/229057068-2fb0b7bd-4759-467f-9f94-366bd9652501.mp4


I looked into this issue, and I think I found the cause of the issue and left a suggestion [here](https://github.com/wordpress-mobile/WordPress-Android/pull/18183/files#r1154147494). I didn't commit the change. You can verify and make the change yourself. Feel free to let me know if you need anything else from me.

File: WordPress/src/main/java/org/wordpress/android/ui/photopicker/MediaPickerLauncher.kt (hunk 1 above)
Comment (irfano) [reply WordPress/src/main/java/org/wordpress/android/ui/photopicker/MediaPickerLauncher.kt]: Thank you so much for testing and finding a critical problem with the implementation, @AjeshRPai! 
I added the support for "Image picker" or "Audio picker" but not for "Image + video + audio picker".
Now, I have added all media support to the media picker: ac21e3c732fa7d3779a8bb185f1710e3535b5d0c.

**Comment by irfano** [comment]
I discovered another issue: the 'Story post' screen was not functioning properly.
I fixed it with the following commit: 898b117a086de961e7e9fbcc3de0014b1d3cb77c.
Additionally, I included it in the testing instructions. However, it can only be tested once https://github.com/Automattic/stories-android/pull/731 is ready.

**Comment by 0nko** [comment]
Hey there! 👋 Looking at this PR, I've been wondering why WPAndroid has its own version of the Media Picker instead of the using the [library](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android), which was designed to be customizable and reusable across different apps.

Is there anything specific that the library doesn't have or why it can't be integrated?

**Comment by irfano** [comment]
> Hey there! 👋 Looking at this PR, I've been wondering why WPAndroid has its own version of the Media Picker instead of the using the [library](https://github.com/wordpress-mobile/WordPress-MediaPicker-Android), which was designed to be customizable and reusable across different apps.
> 
> Is there anything specific that the library doesn't have or why it can't be integrated?

The reason for not using the MediaPicker library previously was that it required Hilt, which was not available when I attempted to [add the Media Picker library to WPAndroid](https://github.com/wordpress-mobile/WordPress-Android/pull/15608).

Now we have Hilt on WPAndroid, and we can add the library. I'll create an issue. Thanks for bringing it up, @0nko.

Btw, I recall we discussed this exact topic before. 😅 (Internal ref: p1637924023002400-slack-C02814L5EGP)

**Comment by 0nko** [comment]
> Btw, I recall we discussed this exact topic before. 😅 (Internal ref: p1637924023002400-slack-C02814L5EGP)

Good memory, it was 2.5 years ago. I completely forgot 😆

File: WordPress/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -2850,6 +2854,10 @@
     <string name="permissions_denied_title">Permissions</string>
     <string name="permissions_denied_message">It looks like you turned off permissions required for this feature.&lt;br/&gt;&lt;br/&gt;To change this, edit your permissions and make sure &lt;strong&gt;%s&lt;/strong&gt; is enabled.</string>
     <string name="permission_storage">Storage</string>
+    <string name="permission_images">Photos and videos</string>
+    <string name="permission_video">Photos and videos</string>
```
Comment (AjeshRPai) [inline WordPress/src/main/res/values/strings.xml]: nip 🔍 : cant we reference the `pemission_images` string here?. Like 

`<string name="permission_video" translatable="false">@string/permission_images </string>`

**Review by AjeshRPai — APPROVED** [review]
Hey @irfano 

I have tested the app for the Media permissions, everything looks good to me 👍🏼 

I have tested all the features by enabling/disabling permissions, and everything works as expected. I was able to select and upload photos and video when given permission and I was shown the message to allow permissions when I disabled the permissions. 

The following features works as expected. 
- Media picker ✅ 
- Media browser ✅ 
- Story ✅ 

Thanks for working on this. I have left a Nitpick for you to address. Feel free to merge it once you had a look at it. I am approving but not merging so that you can have a look at the Nitpick I have left.

File: WordPress/src/main/res/values/strings.xml (hunk 2 above)
Comment (irfano) [reply WordPress/src/main/res/values/strings.xml]: Thanks for the idea. I've done it: 21f997592b97b6302889dfc549f21efaa9e25b7b

**Comment by irfano** [comment]
I have also updated the RELEASE-NOTES.
Thank you all for reviewing. I am now merging.

**Pull request merged**